vllm/model_executor/layers/pooler.py

```diff
@@ -13,6 +14,33 @@ class PoolingType(IntEnum):
     LAST = 0
     ALL = 1
     CLS = 2
+    MEAN = 3
+
+
+@dataclass
+class PoolingConfig:
+    """A class that configures the pooling operation which
+      only applies to sentence-transformers models. 
+      More at: https://www.sbert.net/
+
+    Attributes:
+        pooling_type (str): The type of pooling to use. 
+        normalize (bool): Whether to normalize the pooled data.
+
+    Methods:
+        get_pooling_type(pooling_type_name): Returns the pooling 
+        type enum value corresponding to the given string.
+    """
+
+    def __init__(self, pooling_type: str, normalize: bool):
+        self.pooling_type = self.get_pooling_type(pooling_type)
+        self.normalize = normalize
+
+    def get_pooling_type(self, pooling_type_name: str) -> PoolingType:
+        pooling_types = PoolingType.__dict__.items()
+        return PoolingType(
+            next((value for key, value in pooling_types
+                  if key.lower() in pooling_type_name), PoolingType.CLS))
```

Comment:
To better integrate with #9697, can we allow the user to override this via CLI? The order of priority should be: `EngineArgs` > sentence-transformers > model defaults

Reply:
Working on it!